src/bodies.cpp

```diff
@@ -652,7 +661,7 @@ bool bodies::Box::intersectsRay(const Eigen::Vector3d& origin, const Eigen::Vect
     if (tmax - tmin > detail::ZERO)
     {
       intersections->push_back(tmin * dir + origin);
-      if (count > 1)
+      if (count == 0 || count > 1)
```

Comment:
```suggestion
      if (count >= 1)
```

Reply:
I don't think your suggestion is correct... maybe `count != 1` would look better, but I thought writing it down explicitly like this would be less confusing for future readers of the code

Reply:
You are right, I misread the condition. Maybe, indeed `count != 1` would be the clearer expression.

Reply:
What's the meaning of `count` here? I thought it would be "number of intersections".

Reply:
it's not an output parameter, but an input one - if set to 1, it tells the algorithm it doesn't need to spend time searching for the second intersection if there is some... 

I still think that the way I put the condition is better self-documenting as it explicitly says that in case `count == 0` we also want to process all the intersections